### PR TITLE
refactor(styles): rewrite block styles

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Release Notes

- **New Features**
  - Introduced a new image display component (`PluginBlockItemImg`) with optional checkbox functionality.
  - Enhanced block list display with a new header section and improved item rendering using the `PluginBlockItemImg` component.

- **Improvements**
  - Improved filtering capabilities for selection options in the internationalization binding component.
  - Enhanced visual representation with SVG icons across various components.
  - Streamlined user interactions and error handling in several components, including block management and version selection.
  - Added new props for enhanced functionality in the block list component, allowing for multi-selection and layout customization.
  - Improved layout and error handling in the block group panel.
  - Enhanced user interaction by providing clearer feedback and improved layout for the folder settings interface.
  - Improved responsiveness and usability in the lifecycle management interface.

- **Bug Fixes**
  - Simplified event handling by removing unnecessary tooltip components, enhancing clarity and maintainability.

- **Styling Updates**
  - Adjusted styles for better layout and responsiveness across multiple components, including padding and flexbox adjustments.
  - Enhanced layout properties for improved presentation of block items and user interfaces.
  - Updated styles to improve the visual organization of components, including the block panel and tree structure.

### CODE REVIEW

File: packages/common/component/PluginBlockItemImg.vue [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+export default {
+  components: {
+    TinyCheckbox: Checkbox
+  },
+  props: {
+    item: {
+      type: Object,
+      default: () => {}
+    },
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/common/component/PluginBlockItemImg.vue]: corrected

File: packages/common/component/BlockDescription.vue [hunk 2]
```diff
@@ -1,7 +1,10 @@
 <template>
-  <meta-description class="description" type="warning">
+  <meta-description class="description">
     <template #content>
-      <div @click="openBlockSetting" class="setting-block"><span class="add-icon">+</span>设置区块暴露属性</div>
+      <div @click="openBlockSetting" class="setting-block">
```
Comment (chilingling) [inline packages/common/component/BlockDescription.vue]: 调整一下顺序，事件绑定在最后

File: packages/common/component/PluginBlockItemImg.vue [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      @error="$event.target.src = defaultImg"
+    />
+    <svg-icon v-else class="item-image item-default-img" name="block-default"></svg-icon>
+    <div v-if="showCheckbox && !displayTable" class="top-left">
+      <tiny-checkbox v-model="computedChecked" @change="handleCheck"></tiny-checkbox>
+    </div>
+  </div>
+</template>
+
+<script>
```
Comment (chilingling) [inline packages/common/component/PluginBlockItemImg.vue]: 新组件建议使用 `setup` 风格

File: packages/common/component/PluginBlockList.vue [hunk 4]
```diff
@@ -1,4 +1,10 @@
 <template>
+  <div v-if="blockStyle === 'mini'" class="header">
```
Comment (chilingling) [inline packages/common/component/PluginBlockList.vue]: `mini` 抽取一个常量出来

File: packages/plugins/materials/src/meta/block/src/BlockVersionSelect.vue [hunk 5]
```diff
@@ -1,21 +1,29 @@
 <template>
   <plugin-setting v-if="panel.created" v-show="panel.show" :title="state.title" class="version-list-panel">
     <template #header>
-      <tiny-button type="primary" @click="selectVersion">确定</tiny-button>
+      <!-- <tiny-button type="primary" @click="selectVersion">确定</tiny-button> -->
```
Comment (chilingling) [inline packages/plugins/materials/src/meta/block/src/BlockVersionSelect.vue]: 注释删除一下

File: packages/common/component/BlockDescription.vue (hunk 2 above)
Comment (gene9831) [reply packages/common/component/BlockDescription.vue]: updated

File: packages/common/component/PluginBlockList.vue (hunk 4 above)
Comment (gene9831) [reply packages/common/component/PluginBlockList.vue]: updated

File: packages/plugins/materials/src/meta/block/src/BlockVersionSelect.vue (hunk 5 above)
Comment (gene9831) [reply packages/plugins/materials/src/meta/block/src/BlockVersionSelect.vue]: comment deleted

File: packages/common/component/PluginBlockItemImg.vue (hunk 3 above)
Comment (gene9831) [reply packages/common/component/PluginBlockItemImg.vue]: refactored

File: packages/common/component/PluginBlockItemImg.vue [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    <img
+      v-if="item.screenshot"
+      class="item-image"
+      :src="item.screenshot || defaultImg"
+      draggable="false"
+      @error="$event.target.src = defaultImg"
+    />
+    <svg-icon v-else class="item-image item-default-img" name="block-default"></svg-icon>
+    <div v-if="showCheckbox && !displayTable" class="top-left">
+      <tiny-checkbox v-model="computedChecked" @change="handleCheck"></tiny-checkbox>
```
Comment (chilingling) [inline packages/common/component/PluginBlockItemImg.vue]: 直接用 checked 值是不是会更好？感觉 v-model 加 computed 怪怪的。

`:modelvalue="checked" `

Comment (gene9831) [reply packages/common/component/PluginBlockItemImg.vue]: {48FEB448-D011-4915-B03C-D1D0755ECADD}
不建议直接修改props传进来的值。
~~使用computed的目的是传一个只读的值给checkbox，其他checkbox的api都没有比较合适的~~

Comment (gene9831) [reply packages/common/component/PluginBlockItemImg.vue]: 可以直接使用 `:modelvalue`

**Review by chilingling — DISMISSED** [review]

File: packages/design-core/assets/编辑2.svg
Comment (hexqi) [inline packages/design-core/assets/编辑2.svg:1]: 这个svg要换个名字，不能用中文

**Review by hexqi — COMMENTED** [review]
image
区块这里有个样式问题也一起改下：间距太大

File: packages/design-core/assets/编辑2.svg
Comment (gene9831) [reply packages/design-core/assets/编辑2.svg:1]: @lichunn 批量上传了中文命名的svg，请求她进行更改

Comment (lichunn) [reply packages/design-core/assets/编辑2.svg:1]: 这个图标在已合入的分支改名为data-edit.svg

Comment (gene9831) [reply packages/design-core/assets/编辑2.svg:1]: `编辑2.svg` file removed

**Comment by gene9831** [comment]
> image 区块这里有个样式问题也一起改下：间距太大

已修改
{87BCEA32-51A2-4AB3-9810-46DD05C9EED6}

**Review by hexqi — APPROVED** [review]

**Pull request merged**